cmake/OpenCVDetectCUDA.cmake

```diff
@@ -108,6 +108,11 @@ foreach(ARCH IN LISTS ARCH_LIST)
   set(OPENCV_CUDA_ARCH_FEATURES "${OPENCV_CUDA_ARCH_FEATURES} ${ARCH}")
 endforeach()
 
+string(STRIP "${OPENCV_CUDA_ARCH_BIN}" OPENCV_CUDA_ARCH_BIN)
+if(NOT OPENCV_CUDA_ARCH_BIN VERSION_LESS 53)
+  set(HAVE_CUDA_FP16 1)
+endif()
```

Comment:
`CUDA_ARCH_BIN` is not string, but list. Several CCs may be enabled during the build at a time, e.g. "cmake -DWITH_CUDA=ON -DCUDA_ARCH_BIN="61;75" ...`

Reply:
IMHO, runtime check should be applied instead (using current CUDA device).

Reply:
_Hello @asmorkalov and @opencv-alalek,_

Thank you for the comments!
I've updated the PR, so now it adds a run-time check.
Could you please re-review the changes?